File: src/org/jetbrains/jet/j2k/ast/AnonymousClass.java
package org.jetbrains.jet.j2k.ast;

import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.List;

/**
 * @author ignatov
 */
public class AnonymousClass extends Class {
  public AnonymousClass(List<Member> members) {
    super(new IdentifierImpl("anonClass"),
      Collections.<String>emptySet(),
      Collections.<Element>emptyList(),
      Collections.<Type>emptyList(),
      Collections.<Expression>emptyList(),
      Collections.<Type>emptyList(),
      members
    );
  }

  @NotNull
  @Override
  public String toKotlin() {
    return bodyToKotlin();
  }
}

File: src/org/jetbrains/jet/j2k/ast/AnonymousClass.kt
package org.jetbrains.jet.j2k.ast

import org.jetbrains.jet.j2k.Converter
import org.jetbrains.jet.j2k.ast.types.Type
import java.util.Collections

public open class AnonymousClass(converter : Converter, members : List<Node>)
                                 : Class(converter,
                                         Identifier("anonClass"),
                                         arrayList(),
                                         Collections.emptySet<Modifier>()!!,
                                         Collections.emptyList<Element>()!!,
                                         Collections.emptyList<Type>()!!,
                                         Collections.emptyList<Expression>()!!,
                                         Collections.emptyList<Type>()!!, members) {
    public override fun toKotlin() = bodyToKotlin()
}
